core/jobs.py

```diff
@@ -753,10 +751,6 @@ def get_mapper_param(param_name):
         Raises:
             Exception: The parameter is not associated to this job type.
         """
-        mapper_params = context.get().mapreduce_spec.mapper.params
-        if param_name not in mapper_params:
```

Comment:
I removed these lines as an exception would be thrown by process_and_flush_pending_tasks if there are invalid params and this error msg will never be returned. To explain this I added tests for user query jobs.